lib/scoreboard.rb

```diff
@@ -27,28 +27,28 @@ class Scoreboard
     [10],
     [10],
     [10, 10, 10]
-  ]
+  ].freeze
   def initialize(frames)
-    if frames == LOWEST_SCORE
-      @result = 0
-		elsif
-      frames == HIGHEST_SCORE
-      @result = 300
-		else
-			@result = calculate_score(frames)
-    end
+    @result = case frames
+              when LOWEST_SCORE
+                0
+              when HIGHEST_SCORE
+                300
+              else
+                calculate_score(frames)
+              end
   end
 
   def calculate_score(frames)
     score = 0 # score starts at 0
-    frames.each_with_index do |frame, index| 
-			score += frame.sum # this will add the sum of rolls in each frame to the score
-			if frame [0] == 10 # if frame is strike
-				score += frames[index + 1][0..1].sum # it will add next frame score to the strike score which is 10
-			elsif frame.sum == 10 # if frame is spare
-				score += frames[index + 1][0] # it will add next frame first rolle score to the spare frame score which is 10
-			end
-		end
-		score # return the score
+    frames.each_with_index do |frame, index|
+      score += frame.sum # this will add the sum of rolls in each frame to the score
+      if frame [0] == 10 # if frame is strike
+        score += frames[index + 1][0..1].sum # it will add next frame score to the strike score which is 10
```

Comment:
Have you tried a test with a scoreboard where there are 2 strikes in a row? e.g. [10,0], [10,0], [8,0]. I believe the bonus points for the first strike frame should be the sum of the next two frames i.e. in this example that would be 18 so the total points for the first frame in this example would be 28 (10+18 bonus). However I think your program would calculate the bonus to be 10 . Please do tell me if I've misunderstood though!

Reply:
I agree with @kat-mcbrooks - there's a bug here :)